website/static/js/profile.js

```diff
@@ -468,7 +468,44 @@ var NameViewModel = function(urls, modes, preventUnsaved, fetchCallback) {
         return suffix;
     };
 
+    self.fullNameParsed = function() {
+        let fullName = $.trim(self.full());
+        let fullNameObj = {};
+        fullNameObj.firstName = '';
+        fullNameObj.middleName = '';
+        fullNameObj.lastName = '';
+        fullNameObj.suffix = '';
+
+        fullName = fullName.split(/(\s+)/).filter( function(name) {
+            return name.trim().length > 0;
+        });
+
+        if (fullName.length === 4 && fullName[3].includes('.')) {
```

Comment:
The `.` check is too fragile. Might consider other ways to check for suffixes including string matching a "complete" list of suffixes.